src/inmanta/export.py

```diff
@@ -508,12 +509,23 @@ def deploy_code(self, conn: protocol.SyncClient, tid: uuid.UUID, version: Option
         code_manager = loader.CodeManager()
         LOGGER.info("Sending resources and handler source to server")
 
+        types = set()
+
         # Load both resource definition and handlers
         for type_name, resource_definition in resource.get_resources():
             code_manager.register_code(type_name, resource_definition)
+            types.add(type_name)
 
         for type_name, handler_definition in Commander.get_providers():
             code_manager.register_code(type_name, handler_definition)
+            types.add(type_name)
+
+        # Register all reference and mutator code to all resources. This is very coarse grained and can be optimized once
```

Comment:
it register everything that is loaded, even if not in use. There is something to say for that, but it is very greedy.